### PR TITLE
Fix `quick-pr-diff-options` on GHE

### LINKED ISSUE #5039: `quick-pr-diff-options` hides all other dropdowns on Github enterprise
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.

### Include in this issue:

- [X] Screenshots/video/gif demonstrating the bug, if it’s visual
- [ ] Console errors, if any

### Description

From an update today of Refined Github to 22.11.1, the dropdown filter menus in the 'Files Changed' view on a PR no longer appear:

image

This occurs only on Github Enterprise (3.2.1) and not public Github. 

From inspecting the source, something seems to erroneously apply the `d-lg-none` class to the div containing the dropdowns. Copy-paste of relevant HTML from the enterprise server is below, since this doesn't apply to any public server.

<details>
<summary> Expand</summary>

CSS Selector Path:

```
html body.logged-in.enterprise.env-production.page-responsive.full-width div.application-main div main#js-repo-pjax-container div.container-xl.clearfix.new-discussion-timeline.px-3.px-md-4.px-lg-5 div#repo-content-pjax-container.repository-content div.position-relative.js-review-state-classes.js-suggested-changes-files-tab div#files_bucket.files-bucket.files-next-bucket.clearfix.pull-request-tab-content.is-visible.js-multi-line-comments-enabled div.pr-toolbar.js-sticky.js-position-sticky.d-flex div.diffbar.details-collapse.js-details-container.Details.flex-1.d-flex.flex-items-center.width-full div.flex-auto.min-width-0 div
```

HTML with Refined Github disabled (from one div above):
```html
<div class="flex-auto min-width-0">
                  <div class="show-if-stuck hide-md hide-sm">
                    <h1 class="d-flex text-bold f5">
  <a class="js-issue-title css-truncate css-truncate-target Link--primary width-fit markdown-title" href="#">Update readme with how to start from existing repo</a>
  <span class="gh-header-number color-text-tertiary pl-1">#2</span>
</h1>

                  </div>
                  <div> <!-- LOOK HERE! this div becomes <div class="d-lg-none"> -->
                      

<details class="details-reset details-overlay diffbar-item diffbar-range-menu select-menu hide-sm hide-md hide-lg  ml-0">
  <summary class="btn-link Link--muted select-menu-button" data-hotkey="c" aria-haspopup="menu" role="button">
      Changes from <strong>all commits</strong>
  </summary>

  <details-menu class="select-menu-modal position-absolute" style="z-index: 99;" role="menu">
    <div class="select-menu-header">
      <span class="select-menu-title">Commits</span>
    </div>
    <div class="select-menu-list">
      <a href="/My-Team/sandbox/pull/2/files" class="select-menu-item" role="menuitem">
        <div class="select-menu-item-text">
          <div class="text-emphasized css-truncate css-truncate-target">
            Show all changes
          </div>
          <span class="description">
            1 commit
          </span>
        </div>
      </a>

          <div class="select-menu-item disabled" style="pointer-events: none;">
            <div class="select-menu-item-text">
              <div class="text-bold">
                Show changes since your last review
              </div>
              <span class="description">
                  You haven’t reviewed this pull request yet
              </span>
            </div>
          </div>

      <div class="select-menu-divider">
        Select commit
      </div>

      <div data-range-url="/My-Team/sandbox/pull/2/files/$range">
          <a href="/My-Team/sandbox/pull/2/commits/2d35a7ffce6fc9e2672fd1757bb498815f7cbb19" class="select-menu-item " role="menuitem" data-commit="2d35a7f">

            <div class="select-menu-item-text">
              <code class="float-right">2d35a7f</code>
              <div class="text-emphasized css-truncate css-truncate-target">
                Update readme with how to start from existing repo
              </div>
              <span class="description">
                jmerdich <relative-time datetime="2021-05-03T18:23:01Z" class="no-wrap" title="May 3, 2021, 2:23 PM EDT">on May 3</relative-time>
              </span>
            </div>
          </a>
      </div>
    </div>
  </details-menu>
</details>


                      <details class="diffbar-item details-reset details-overlay select-menu d-inline-block position-relative js-file-filter hide-sm">
  <summary class="Link--muted select-menu-button" data-ga-click="Pull Requests, open file types filter" aria-haspopup="menu" role="button">
    <span class="select-menu-title">
      <strong class="js-file-filter-text css-truncate css-truncate-target">
        File filter
      </strong>
    </span>
  </summary>

  <details-menu class="select-menu-modal position-absolute left-0" style="z-index: 99;">
    <div class="select-menu-header d-flex flex-justify-between">
      <span class="select-menu-title">Filter file types</span>
    </div>
    <div class="select-menu-list" role="menu">
      <form class="js-file-filter-form">
        <div class="p-2">
            <label class="pl-1 mb-1 d-block" style="cursor:pointer;" role="menuitemcheckbox" tabindex="0">
              <input class="js-diff-file-type-option" style="cursor:pointer;" type="checkbox" value=".md" name="file-filters[]" data-non-deleted-files-count="1" data-deleted-files-count="0" checked="">
              .md
              <span class="text-normal js-file-type-count" data-non-deleted-file-count-markup="(1)" data-all-file-count-markup="(1)">
                (1)
              </span>
            </label>
          <div class="ml-1 mt-1">
            <label style="cursor:pointer;">
              <input type="checkbox" class="js-file-filter-select-all" hidden="">
              <span class="color-text-secondary no-underline text-normal js-file-filter-select-all-container" data-select-all-markup="Select all 1 file type" data-all-selected-markup="All 1 file type selected">
                All 1 file type selected
              </span>
            </label>
          </div>
        </div>


        <div>
          <div class="select-menu-header border-y color-border-secondary d-flex flex-justify-between">
            <span class="select-menu-title">Filter viewed files</span>
          </div>
          <div class="ml-1 p-2">
            <label style="cursor:pointer;" role="menuitemcheckbox" tabindex="0">
              <input type="checkbox" class="js-viewed-files-toggle" value="true" checked="">
                Hide viewed files
            </label>
          </div>
        </div>

        <div class="select-menu-header d-flex flex-justify-between border-top">
          <span class="select-menu-title">Filter by context</span>
        </div>
        <div class="select-menu-list" role="menu">
          <div class="p-2">
            <div class="d-flex">
              <label class="pl-1 mb-1" style="cursor:pointer;" role="menuitemcheckbox" tabindex="0">
                <input class="js-manifests-option" type="checkbox" value="true" name="manifests">
                Manifest files
              </label>
            </div>
          </div>
        </div>
      </form>
    </div>

  </details-menu>
</details>

                        <div class="subset-files-tab js-reset-filters px-sm-3 px-0 mr-sm-0 mr-3" hidden="">
  <a class="text-bold color-text-link no-underline ml-sm-2 ml-0" href="/My-Team/sandbox/pull/2/files">
    <svg class="octicon octicon-x color-text-secondary" height="14" viewBox="0 0 16 16" version="1.1" width="14" aria-hidden="true"><path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"></path></svg> <span>Clear filters</span>
  </a>
</div>

                      
<div class="position-relative diffbar-item ml-0">
  <details class="details-reset details-overlay toc-select js-conversations-details select-menu ml-0 ml-sm-3">
    <summary class="btn-link Link--muted select-menu-button" data-ga-click="Pull Requests, open view comments menu, type:semantic" aria-haspopup="menu" role="button">
      <strong class="js-conversation-menu-button">Conversations</strong>
    </summary>
    <details-menu class="select-menu-modal position-absolute" style="z-index: 99;" id="conversations-menu" src="/My-Team/sandbox/pull/2/conversations_menu" preload="" role="menu">
      <include-fragment>
        <div class="select-menu-blankslate select-menu-error">
          Failed to load comments. <button type="button" class="btn-link js-toc-retry">Retry</button>
        </div>
        <div class="select-menu-loading-overlay d-flex flex-items-center">
          <svg style="box-sizing: content-box; color: var(--color-icon-primary);" viewBox="0 0 16 16" fill="none" data-view-component="true" width="32" height="32" class="flex-1 anim-rotate">
  <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke"></circle>
  <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-width="2" stroke-linecap="round" vector-effect="non-scaling-stroke"></path>
</svg>
        </div>
      </include-fragment>
    </details-menu>
  </details>
</div>

                      
<details class="details-reset details-overlay diffbar-item toc-select select-menu ml-0 ml-sm-3">
  <summary class="btn-link Link--muted select-menu-button" data-hotkey="t" data-ga-click="Pull Requests, open table of contents, type:semantic" aria-haspopup="menu" role="button">
    <strong>Jump to</strong>
  </summary>
  <details-menu class="select-menu-modal position-absolute" style="z-index: 99;" src="/My-Team/sandbox/pull/2/show_toc?base_sha=66994d362e672a6e38df20d6ce33c9bb16a4c35e&amp;sha1=66994d362e672a6e38df20d6ce33c9bb16a4c35e&amp;sha2=2d35a7ffce6fc9e2672fd1757bb498815f7cbb19" preload="" role="menu">
      <div class="select-menu-header">
        <span class="select-menu-title">
          Jump to file
        </span>
      </div>
      <include-fragment>
        <div class="select-menu-blankslate select-menu-error">
          Failed to load files. <button type="button" class="btn-link js-toc-retry">Retry</button>
        </div>
        <div class="select-menu-loading-overlay" aria-label="Loading">
          <svg style="box-sizing: content-box; color: var(--color-icon-primary);" viewBox="0 0 16 16" fill="none" data-view-component="true" width="32" height="32" class="anim-rotate">
  <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke"></circle>
  <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-width="2" stroke-linecap="round" vector-effect="non-scaling-stroke"></path>
</svg>
        </div>
      </include-fragment>
  </details-menu>
</details>

                      <details class="diffbar-item details-reset details-overlay position-relative text-center">
  <summary role="button" data-view-component="true" class="btn-link">
  
  
    <svg class="octicon octicon-gear color-text-secondary" aria-label="Diff settings" viewBox="0 0 16 16" version="1.1" width="16" height="16" role="img"><path fill-rule="evenodd" d="M7.429 1.525a6.593 6.593 0 011.142 0c.036.003.108.036.137.146l.289 1.105c.147.56.55.967.997 1.189.174.086.341.183.501.29.417.278.97.423 1.53.27l1.102-.303c.11-.03.175.016.195.046.219.31.41.641.573.989.014.031.022.11-.059.19l-.815.806c-.411.406-.562.957-.53 1.456a4.588 4.588 0 010 .582c-.032.499.119 1.05.53 1.456l.815.806c.08.08.073.159.059.19a6.494 6.494 0 01-.573.99c-.02.029-.086.074-.195.045l-1.103-.303c-.559-.153-1.112-.008-1.529.27-.16.107-.327.204-.5.29-.449.222-.851.628-.998 1.189l-.289 1.105c-.029.11-.101.143-.137.146a6.613 6.613 0 01-1.142 0c-.036-.003-.108-.037-.137-.146l-.289-1.105c-.147-.56-.55-.967-.997-1.189a4.502 4.502 0 01-.501-.29c-.417-.278-.97-.423-1.53-.27l-1.102.303c-.11.03-.175-.016-.195-.046a6.492 6.492 0 01-.573-.989c-.014-.031-.022-.11.059-.19l.815-.806c.411-.406.562-.957.53-1.456a4.587 4.587 0 010-.582c.032-.499-.119-1.05-.53-1.456l-.815-.806c-.08-.08-.073-.159-.059-.19a6.44 6.44 0 01.573-.99c.02-.029.086-.075.195-.045l1.103.303c.559.153 1.112.008 1.529-.27.16-.107.327-.204.5-.29.449-.222.851-.628.998-1.189l.289-1.105c.029-.11.101-.143.137-.146zM8 0c-.236 0-.47.01-.701.03-.743.065-1.29.615-1.458 1.261l-.29 1.106c-.017.066-.078.158-.211.224a5.994 5.994 0 00-.668.386c-.123.082-.233.09-.3.071L3.27 2.776c-.644-.177-1.392.02-1.82.63a7.977 7.977 0 00-.704 1.217c-.315.675-.111 1.422.363 1.891l.815.806c.05.048.098.147.088.294a6.084 6.084 0 000 .772c.01.147-.038.246-.088.294l-.815.806c-.474.469-.678 1.216-.363 1.891.2.428.436.835.704 1.218.428.609 1.176.806 1.82.63l1.103-.303c.066-.019.176-.011.299.071.213.143.436.272.668.386.133.066.194.158.212.224l.289 1.106c.169.646.715 1.196 1.458 1.26a8.094 8.094 0 001.402 0c.743-.064 1.29-.614 1.458-1.26l.29-1.106c.017-.066.078-.158.211-.224a5.98 5.98 0 00.668-.386c.123-.082.233-.09.3-.071l1.102.302c.644.177 1.392-.02 1.82-.63.268-.382.505-.789.704-1.217.315-.675.111-1.422-.364-1.891l-.814-.806c-.05-.048-.098-.147-.088-.294a6.1 6.1 0 000-.772c-.01-.147.039-.246.088-.294l.814-.806c.475-.469.679-1.216.364-1.891a7.992 7.992 0 00-.704-1.218c-.428-.609-1.176-.806-1.82-.63l-1.103.303c-.066.019-.176.011-.299-.071a5.991 5.991 0 00-.668-.386c-.133-.066-.194-.158-.212-.224L10.16 1.29C9.99.645 9.444.095 8.701.031A8.094 8.094 0 008 0zm1.5 8a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zM11 8a3 3 0 11-6 0 3 3 0 016 0z"></path></svg>
    <div class="dropdown-caret color-text-secondary"></div>

  

</summary>  <div class="Popover js-diff-settings mt-2 pt-1" style="left: -108px">
    <div class="Popover-message text-left p-3 mx-auto Box color-shadow-large">
      <!-- '"` --><!-- </textarea></xmp> --><form action="/My-Team/sandbox/pull/2/files" accept-charset="UTF-8" method="get">
        <h4 class="mb-2">Always</h4>
        <div class="BtnGroup d-flex flex-content-stretch js-diff-style-toggle">
  <label class="flex-auto btn btn-sm BtnGroup-item text-center ">
    <input class="sr-only" value="unified" name="diff" type="radio">
     Unified
  </label>
  <label class="flex-auto btn btn-sm BtnGroup-item text-center selected">
    <input class="sr-only" value="split" name="diff" type="radio" checked="">
     Split
  </label>
</div>

        <h4 class="mb-2 mt-3">Just for now</h4>
        <input type="checkbox" name="w" value="1" id="whitespace-cb">
        <label for="whitespace-cb" class="text-normal">Hide whitespace changes</label>
        <button class="btn btn-primary btn-sm col-12 mt-3" type="submit">Apply and reload</button>
</form>    </div>
  </div>
</details>

                          <div class="js-socket-channel js-updatable-content js-pull-refresh-on-pjax" data-channel="eyJjIjoicHVsbF9yZXF1ZXN0OjIyNzYxIiwidCI6MTYzNTg3MDkwNn0=--a160425a268932d5fdd6f32dee1a32ce7fbf3f74b401617860dccae1d56c964a" data-url="/My-Team/sandbox/pull/2/show_partial_comparison?base_commit_oid=66994d362e672a6e38df20d6ce33c9bb16a4c35e&amp;end_commit_oid=2d35a7ffce6fc9e2672fd1757bb498815f7cbb19&amp;partial=pull_requests%2Fstale_comparison&amp;start_commit_oid=66994d362e672a6e38df20d6ce33c9bb16a4c35e" data-gid="MDExOlB1bGxSZXF1ZXN0MjI3NjE="></div>

                  </div>
                </div>
```
</details>

### Example URL

https://github.com/refined-github/refined-github/pull/4918/files

### Browser(s) used

Firefox 93.0 (Win10)

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #5039 

## Test URLs

 * [Files tab](https://github.com/refined-github/sandbox/pull/4/files)
 * [Single commit](https://github.com/refined-github/refined-github/commit/49fd6261e21d69bd62094bf0b38783a7e06020d1)

## Screenshot

n/a

### CODE REVIEW

File: source/features/quick-pr-diff-options.tsx [hunk 1]
```diff
@@ -74,8 +74,10 @@ function createWhitespaceButton(): HTMLElement {
 }
 
 function initPR(): false | void {
-	// TODO [2022-05-01]: Remove `.js-diff-settings` from the selector (kept for GHE)
-	const originalToggle = select('.js-diff-settings, [aria-label="Diff settings"]')!.closest('details')!.parentElement!;
+	// TODO [2022-05-01]: Remove GHE code
+	const originalToggle = pageDetect.isEnterprise()
+		? select('.js-diff-settings')!.closest('details')!
+		: select('[aria-label="Diff settings"]')!.closest('details')!.parentElement!;
```
Comment (cheap-glitch) [inline source/features/quick-pr-diff-options.tsx:80]: #5022 added a `.parentElement` which broke GHE

**Review by cheap-glitch — COMMENTED** [review]
@busches could you test this on both URLs please? Thanks!

**Comment by busches** [comment]
Testing on GHE 3.2.3 and it looks fine:
image
Pressing ignore whitespace triggers it correctly, but the button is gone to toggle it off:
image
Diff view button works fine:
image

File: source/features/quick-pr-diff-options.tsx [hunk 2]
```diff
@@ -74,8 +74,10 @@ function createWhitespaceButton(): HTMLElement {
...
 
 	if (!isHidingWhitespace()) {
```
Comment (busches) [inline source/features/quick-pr-diff-options.tsx:82]:
```suggestion
		if (!isHidingWhitespace() || pageDetect.isEnterprise()) {
```
From what I can tell, GH natively adds this 'Show whitespace' button back and GHE does not, so we want to add it back all the time on GHE.

File: source/features/quick-pr-diff-options.tsx [hunk 3]
```diff
@@ -74,10 +74,12 @@ function createWhitespaceButton(): HTMLElement {
 }
 
 function initPR(): false | void {
-	// TODO [2022-05-01]: Remove `.js-diff-settings` from the selector (kept for GHE)
-	const originalToggle = select('.js-diff-settings, [aria-label="Diff settings"]')!.closest('details')!.parentElement!;
+	// TODO [2022-05-01]: Remove GHE code
```
Comment (fregante) [inline source/features/quick-pr-diff-options.tsx:77]: Should be on the GHE line, otherwise it's not clear which one is "GHE code"

**Review by fregante — APPROVED** [review]

**Pull request merged**